### ISSUE TITLE
When to use `CONST` prior?

### ISSUE THREAD

**Issue by dafeda** [equinor/ert#2627]
When all coefficients in the poly example are set to `CONST 1.0`, analysis update is not run.

I think this makes sense as there is perhaps no point in running an update step if all columns in the parameter matrix `A` are the same.

What is the intended use of the `CONST` keyword?

I think I've tracked down the part of code that disables rows that are too similar.
This happens in the `enkf_analysis_deactivate_outliers` function which calculated the standard deviation of a row, and deactivates it if it's too low.

https://github.com/equinor/ert/blob/34b398170e9aca51fb5a9a94e9dd506d6a4db2de/libres/lib/enkf/enkf_analysis.cpp#L150-L159

Would be great to get your thoughts on this one @joakim-hove .

**Comment by joakim-hove** [comment]
You can call me on 92 68 57 04 - or I Will answer in writing tomorrow.

**Comment by dafeda** [comment]
> You can call me on 92 68 57 04 - or I Will answer in writing tomorrow.

Thanks!
No hurry, you can write whenever convenient.

**Comment by joakim-hove** [comment]
> I think I've tracked down the part of code that disables rows that are too similar.

This is a misunderstanding. The code which disables values that are too similar targets simulated values (i.e. measurements) which should be compared with observations. Measurements and observations are *input* to the update algorithm. Whereas the `CONST` distribution which is the topic of this is question is about *parameters* which are updated by the update algorithm.

So - what is the point of having a `CONST` distribution for a scalar parameter? It is not terribly important, and plays no role in a finalized setup for running ERT, but in my opinion it has a value for the interactive workflows while working with ERT. The scenarios where the `CONST`distribution makes sense are varieties of:

A model has been parameterized with uncertain parameters `P1, P2` and `P3` - after an initial run you wonder: "Maybe I should not vary `P2`?" - then it will be much faster to use the `CONST` distribution for the initial test, than to edit various files. 

Mind you: the parameter distributions for the `GEN_KW` keyword are not really prior distributions, but rather transformations - which will transform a `N(0,1)` distributions to the the target distribution. See: https://github.com/equinor/ert/blob/main/libres/lib/enkf/trans_func.cpp

**Comment by dafeda** [comment]
Thanks for the explanation.

From my understanding, the simulated values (i.e. measurements) will in the poly_example case be the same in all realizations if parameters are defined as `CONST`.

`enkf_analysis_deactivate_outliers` will then deactivate everything in `meas_data` which will cause the condition in the if-statment below to be false and hence no analysis update.

https://github.com/equinor/ert/blob/f957974e55fbbbe91389b2eacfa7bb7e1599b2b4/libres/lib/analysis/update.cpp#L738-L739

I'm probably misunderstanding again, but hey, I learn by a slow form of gradient descent.

**Comment by joakim-hove** [comment]
> From my understanding, the simulated values (i.e. measurements) will in the poly_example case be the same in all realizations if parameters are defined as CONST.

Ehhh - if *all* parameters in the poly_example are distributed according to the `CONST` "distribution" then yes all simulated values will be equal, and then the `enkf_analysis_deactivate_outliers()` will deactivate observations as you indicate.

**Comment by dafeda** [comment]
> Ehhh - if _all_ parameters in the poly_example are distributed according to the `CONST` "distribution" then yes all simulated values will be equal, and then the `enkf_analysis_deactivate_outliers()` will deactivate observations as you indicate.

The "Ehhh" is not necessary, but thanks.

**Comment by joakim-hove** [comment]
Considering the poly example. That example has two files for parameters:
**coeff_priors:**
```
COEFF_A NORMAL 0.5 0.5 
COEFF_B NORMAL 1.0 1.0
COEFF_C NORMAL 2.5 2.5
```
**coeff.tmpl**:
```
{
  "a": <COEFF_A>,
  "b": <COEFF_B>,
  "c": <COEFF_C>
}
```
and the forward model `poly_eval.py` expects to load a  generated json file with three parameters `a`, `b` and `c`. If you at some stage in the update process decide that you want to try out a fixed value of `b == 2.0` it is faster and simpler to just update the single file `coeff_prior` to
```
COEFF_A NORMAL 0.5 0.5 
COEFF_B CONST    2.0
COEFF_C NORMAL 2.5 2.5
```
than updating all three files. If you in the concluded that `b` should indeed be fixed at e.g. the value 2.0 you should probably update the files in question and not drag along a parameter which is distributed with `CONST` distribution.
